### ISSUE TITLE
PR's created immediately even when stabilityDays=X and prCreation="not-pending" are set

### ISSUE THREAD

**Issue by rvillane** [renovatebot/config-help#572]
### Which Renovate are you using?
Renovate Open Source CLI

### Which platform are you using?
GitHub Enterprise

### What would you like to do?
I have the following configuration in place

```
module.exports = {
  endpoint: 'https://my.ghe.com/api/v3/',
  platform: 'github',
  logFile: '/dist/renovate.log',
  onboarding: true,
  onboardingConfig: {
    extends: ['config:base'],
  },
  autodiscover: true,
  autodiscoverFilter: 'MyOrg/*',
  requireConfig: true,
  automerge: false,
  extends: ['group:allNonMajor'],
  enabledManagers: ['npm'],
  rangeStrategy: 'pin',
  recreateClosed: true,
  stabilityDays: 7,
  prCreation:'not-pending',
  labels: [
    'renovate'
  ],
};
```

And according to documentation https://docs.renovatebot.com/configuration-options/#suppress-branchpr-creation-for-x-days, if stabilityDays=X and prCreation="not-pending" settings are provided, the PR won't be created until the period defined in stabilityDays ha passed, but I'm getting PR's created as soon as new dependency version become available, any ideas ?

**Comment by rarkins** [comment]
For which package manager?

**Comment by rvillane** [comment]
NPM

**Comment by rarkins** [comment]
That shouldn't be happening. Can you reproduce in a github.com repository? e.g. create a repo with a single dependency that's out of date and has received an update in the last day or two, use your same settings (in `renovate.json`), and see if the PR is created?

**Comment by rvillane** [comment]
Please look at https://github.com/rvillane/renovate-pinning/pull/12

NodeMailer 6.4.5 was released 3 days ago
mongoDB 3.5.5 was released 2 days ago

Enclosed is the log in debug mode

**Comment by rvillane** [comment]
[debug-renovate-github.com.txt](https://github.com/renovatebot/config-help/files/4329845/debug-renovate-github.com.txt)

**Comment by rvillane** [comment]
I think the PR and branch existed already, however I just added Nodemailer and MongoDB dependencies to the package.json file and re-run renovate.

**Comment by rarkins** [comment]
It doesn't seem like a valid use case. The setting you're talkinga bout controls the creation of PR, but your PR already existed:

```
DEBUG: getBranchPr(renovate/all-minor-patch) (repository=rvillane/renovate-pinning, dependencies=@google-cloud/monitoring,@google-cloud/pubsub,commander,mongodb,nodemailer, branch=renovate/all-minor-patch)
DEBUG: findPr(renovate/all-minor-patch, null, open) (repository=rvillane/renovate-pinning, dependencies=@google-cloud/monitoring,@google-cloud/pubsub,commander,mongodb,nodemailer, branch=renovate/all-minor-patch)
DEBUG: Found PR #12 (repository=rvillane/renovate-pinning, dependencies=@google-cloud/monitoring,@google-cloud/pubsub,commander,mongodb,nodemailer, branch=renovate/all-minor-patch)
```

**Comment by rvillane** [comment]
Ok, I can merge that PR and wait for new updates and create a new new PR, might take a few days.

**Comment by rarkins** [comment]
Behaviour is also undetermined when you are grouping like that. Do you want the PR to be delayed forever as long as one of the dependencies within it is less than 7 days old?

**Comment by rvillane** [comment]
yes, that sounds like the expected behavior on my end.

I actually tried with mongoose and they released a change 4 days ago but a new PR was created.
https://github.com/rvillane/renovate-pinning/pull/13/files

Perhaps, I was just not clear how this feature is intended to work

**Comment by rvillane** [comment]
before this issue is closed, can you please provide some extra detail about how this feature is supposed to work ?

**Comment by rarkins** [comment]
The original goal of `prCreation="not-pending"` was just so that you don't get a PR that's still running tests. i.e. "don't raise it until it's either passed or failed".

`stabilityDays` is designed as a status check to discourage brand new updates from being merged right away. When combined with the `prCreation="not-pending"` setting, ideally it should hold back on creating the PR until the update reaches the desired stability.

Once you start combining multiple packages into a group, the usefulness of this settings combination decreases - it might take a long time until all the updates are X days old, and then once the PR's created then it won't take long until one of the updates turns the status check to pending again.

If you have a single update in a PR, and that branch/PR is created when the update hasn't reached stability days, then it indicates a bug.

**Comment by ibratoev** [comment]
I hit the same problem. In my case, I have a group for all dev dependencies and they are a lot. The PR gets updated all the time and the status check is constantly failed. That happens even with `stabilityDays: 3`.

What do you think about having an option to have separate stable and unstable branches/PRs?
Or an option to ignore the unstable updates and show them only as reference in the master issue?

**Comment by ibratoev** [comment]
@rarkins Do you accept PRs for such issues? Should I prepare a proposal somewhere to get that moving?

**Comment by rarkins** [comment]
@ibratoev please raise an issue in the main repo to discuss before starting an PR. There would be quite a few edge cases needing to be covered before any PR could be merged, so best to discuss them before writing any code that would then end up needing changing later.

I will close this issue for now, as there doesn't seem to be any further config help required.
